src/configure/helper/commonHelper.ts

```diff
@@ -63,4 +67,36 @@ export function stringCompareFunction(a: string, b: string): number {
         return 1;
     }
     return 0;
+}
+
+export async function executeFunctionWithRetry(func: () => Promise<any>, retryCount: number = 100, retryTimeoutInSec: number = 300, retryIntervalTimeInSec: number = 2, errorMessage?: string): Promise<any> {
+    let internalErrorMessage = '';
+    let functionExecutionPromise = Q.Promise(async (resolve, reject) => {
+        let result = null;
+        for (;retryCount > 0; retryCount--) {
+            try {
+                result = await func();
+                resolve(result);
+                break;
+            }
+            catch (error) {
+                internalErrorMessage = errorMessage ? errorMessage.concat(' Internal Error: ', JSON.stringify(error)) : JSON.stringify(error);
+                setTimeout(() => {
+                    // do nothing
+                },
+                retryIntervalTimeInSec * 1000);
+            }
+        }
+
+        if (retryCount <= 0 && !result) {
```

Comment:
why fail is result is null? A legitimate call can return null. Does restClient.sendRequest return null in case of non-200 status code?

Reply:
Changed it to return result if promise resolves and throwing otherwise